src/cmd/restart.go

```diff
@@ -16,21 +17,33 @@ func RestartSpotify(flags ...string) {
 
 	switch runtime.GOOS {
 	case "windows":
-		exec.Command("taskkill", "/F", "/IM", "spotify.exe").Run()
-		if isAppX {
-			ps, _ := exec.LookPath("powershell.exe")
-			exe := filepath.Join(os.Getenv("LOCALAPPDATA"), "Microsoft", "WindowsApps", "Spotify.exe")
-			flags = append([]string{"-NoProfile", "-NonInteractive", `& "` + exe + `" --app-directory="` + appDestPath + `"`}, flags...)
-			exec.Command(ps, flags...).Start()
-		} else {
-			exec.Command(filepath.Join(spotifyPath, "spotify.exe"), flags...).Start()
+		isRunning := exec.Command("tasklist", "/FI", "ImageName eq spotify.exe")
+		result, _ := isRunning.Output()
+		if !bytes.Contains(result, []byte("No tasks are running")) {
+			exec.Command("taskkill", "/F", "/IM", "spotify.exe").Run()
+			if isAppX {
+				ps, _ := exec.LookPath("powershell.exe")
+				exe := filepath.Join(os.Getenv("LOCALAPPDATA"), "Microsoft", "WindowsApps", "Spotify.exe")
+				flags = append([]string{"-NoProfile", "-NonInteractive", `& "` + exe + `" --app-directory="` + appDestPath + `"`}, flags...)
+				exec.Command(ps, flags...).Start()
+			} else {
+				exec.Command(filepath.Join(spotifyPath, "spotify.exe"), flags...).Start()
+			}
 		}
 	case "linux":
-		exec.Command("pkill", "spotify").Run()
-		exec.Command(filepath.Join(spotifyPath, "spotify"), flags...).Start()
+		isRunning := exec.Command("pgrep", "spotify")
+		_, err := isRunning.Output()
+		if err == nil {
+			exec.Command("pkill", "spotify").Run()
+			exec.Command(filepath.Join(spotifyPath, "spotify"), flags...).Start()
+		}
 	case "darwin":
-		exec.Command("pkill", "Spotify").Run()
-		flags = append([]string{"-a", "/Applications/Spotify.app"}, flags...)
-		exec.Command("open", flags...).Start()
+		isRunning := exec.Command("pgrep", "Spotify")
```

Comment:
pgrep is bugged on macOS, try using `ps aux | grep "Spotify" | grep -v grep`

Reply:
Just wanted a quick check, if I were to change it, would it be like so? I apologize for I am not familiar with the language
```suggestion
		isRunning := exec.Command("ps aux | grep Spotify | grep -v grep")
```
The command works on the terminal but currently I am unable to test with the executable for I don't have my hands on a Mac

Reply:
Will check later on macOS.

Reply:
Much appreciated!

Reply:
After doing some research, the suggestion above is an invalid command and will return an error. I've found an alternative solution: 
```suggestion
		isRunning := exec.Command("grep", "Spotify")
		ps := exec.Command("ps", "aux")
		pipe, _ := ps.StdoutPipe()
		defer pipe.Close()
		isRunning.Stdin = pipe
		ps.Start()
```
It should be similar to `ps aux | grep "Spotify"` and should work as expected.

Reply:
After doing some research I've found out that my suggestion above is an invalid command but I've managed to test my initial changes on a Mac and can confirm `pgrep` works as expected, so is there any chance you might reconsider it?
![image](https://user-images.githubusercontent.com/77577746/165693092-d39504e1-f64b-41de-8dd1-686223416ae5.png)

Reply:
I said it's bugged out. Not that it doesn't work. You are on the latest macOS anyway, a lot of users use older versions. A lot of people complain that it doesn't work properly and recommend using brew's formula. We can't force people to install brew formula to have it working properly

Reply:
Alright, understood. I'll try to find a better workaround for it.

Reply:
Committed changes confirmed to be working on macOS
![image](https://user-images.githubusercontent.com/77577746/165710533-606513b4-bc06-4b61-986f-236ea1c9ad32.png)